osbenchmark/workload/params.py

```diff
@@ -200,23 +200,27 @@ def __init__(self, workload, params, **kwargs):
                         if "settings" in body:
                             # merge (and potentially override)
                             body["settings"].update(settings)
+                            self.validate_index_codec(body["settings"])
                         else:
                             body["settings"] = settings
                     elif not body and settings:
                         body = {
                             "settings": settings
                         }
-
                     self.index_definitions.append((idx.name, body))
         else:
             try:
                 # only 'index' is mandatory, the body is optional (may be ok to create an index without a body)
                 idx = params["index"]
                 body = params.get("body")
+                if body and "settings" in body:
+                    self.validate_index_codec(body["settings"])
                 if isinstance(idx, str):
                     idx = [idx]
                 for i in idx:
                     self.index_definitions.append((i, body))
+            except ValueError as e:
+                raise exceptions.InvalidSyntax(f"Please set value properly for create-index operation. {e}")
```

Comment:
nit: "set the value properly for the create-index operation".

Also, the other error messages in this file don't seem to be printing out the exception, consider making the behavior consistent.
```
                raise exceptions.InvalidSyntax("Please set the property 'index' for the create-index operation")
                raise exceptions.InvalidSyntax("Please set the property 'data-stream' for the create-data-stream operation")
                raise exceptions.InvalidSyntax("Please set the properties 'template' and 'body' for the create-index-template operation")
```

Reply:
I've added the "the"s to the error message. I added it to the following exceptions but noticed it didn't add much value since it would just append the property that's already specified in the message (For example it'd look like `Please set the property 'index' for the create-index operation. 'index'`)